### PR TITLE
ci(php): Add PHP8 support.

### PR DESCRIPTION
**Pull request by ptondereau**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
         }
     ],
     "require": {
-        "php": "^7.3",
```
Comment (nunomaduro) [inline composer.json:19]: Can you replace by `^7.3 || ^8.0`?

**Review by nunomaduro — CHANGES_REQUESTED** [review]

**Comment by ptondereau** [comment]
@nunomaduro tests seems not working on PHP8 with prefer-lowest (see CI on first commit)> Seems to be related to a Carbon issue. What do you prefer?

**Comment by nunomaduro** [comment]
@owenvoke any idea here?

**Comment by ptondereau** [comment]
Leaving this as it for now because I need to investigate on Box project side: PHAR fails to be built and it raises an exception when moving it.

**Comment by ptondereau** [comment]
Catched it:

```
/app # php application app:build --build-version=0.0 -v

                                                
                Building process                
                                                

   1. Moving application to production mode: ✔


Box Requirements Checker
========================

> Using PHP 8.0.0-dev
> PHP is using the following php.ini file:
  WARNING: No configuration file (php.ini) used by PHP!

> Checking Box requirements:
  E.....


                                                                                
 [ERROR] Your system is not ready to run the application.                       
                                                                                

Fix the following mandatory requirements:
=========================================

 * The application requires the version "^7.2" or greater.


```

**Comment by ptondereau** [comment]
Closing for now as Box project should allow accepting PHP 8.0 (make an issue https://github.com/box-project/box/issues/486) because of requirements checker. I tried a hack with `config.platform.php` in `composer.json` set to `8.0.0` or `8.0.0-dev`.
I recommend to wait until Box support PHP8

**Pull request closed without merging**